cloud/amazon/cloudformation.py

```diff
@@ -255,148 +343,136 @@ def main():
 
     # check the status of the stack while we are creating/updating it.
     # and get the outputs of the stack
 
     if state == 'present' or update:
-        stack = invoke_with_throttling_retries(cfn.describe_stacks,stack_name)[0]
-        for output in stack.outputs:
-            stack_outputs[output.key] = output.value
-        result['stack_outputs'] = stack_outputs
-        stack_resources = [] 
-        for res in cfn.list_stack_resources(stack_name):
+        stack = get_stack_facts(cfn, stack_params['StackName'])
+        for output in stack.get('Outputs', []):
+            result['stack_outputs'][output['OutputKey']] = output['OutputValue']
+        stack_resources = []
+        reslist = cfn.list_stack_resources(StackName=stack_params['StackName'])
+        for res in reslist.get('StackResourceSummaries', []):
             stack_resources.append({
-                "last_updated_time": res.last_updated_time,
-                "logical_resource_id": res.logical_resource_id,
-                "physical_resource_id": res.physical_resource_id,
-                "status": res.resource_status,
-                "status_reason": res.resource_status_reason,
-                "resource_type": res.resource_type })
+                "logical_resource_id": res['LogicalResourceId'],
+                "physical_resource_id": res['PhysicalResourceId'],
+                "resource_type": res['ResourceType'],
+                "last_updated_time": res['LastUpdatedTimestamp'],
+                "status": res['ResourceStatus'],
+                "status_reason": res.get('ResourceStatusReason') # can be blank, apparently
+            })
         result['stack_resources'] = stack_resources
 
     # absent state is different because of the way delete_stack works.
     # problem is it it doesn't give an error if stack isn't found
     # so must describe the stack first
 
     if state == 'absent':
+        #result = {}
         try:
-            invoke_with_throttling_retries(cfn.describe_stacks,stack_name)
-            operation = 'DELETE'
-        except Exception as err:
-            error_msg = boto_exception(err)
-            if 'Stack:%s does not exist' % stack_name in error_msg:
+            stack = get_stack_facts(cfn, stack_params['StackName'])
+            if not stack:
                 result = dict(changed=False, output='Stack not found.')
             else:
-                module.fail_json(msg=error_msg)
-        if operation == 'DELETE':
-            cfn.delete_stack(stack_name)
-            result = stack_operation(cfn, stack_name, operation)
+                cfn.delete_stack(StackName=stack_params['StackName'])
+                result = stack_operation(cfn, stack_params['StackName'], 'DELETE')
+        except Exception as err:
+            ei = sys.exc_info()
+            error_msg = boto_exception(err)
+            module.fail_json(msg=error_msg + ' ' + str(traceback.format_exception(*ei)))
```

Comment:
Instead of stringifying the traceback, use the `exception` kwarg to `fail_json`. It takes the formatted traceback (which doesn't require an arg inside an exception handler). 

```
except Exception as err:
    module.fail_json(msg=boto_exception(err), exception=traceback.format_exc())
```